plugins/woocommerce-admin/client/subscriptions/index.tsx

```diff
@@ -0,0 +1,344 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	createInterpolateElement,
+	useEffect,
+	useState,
+} from '@wordpress/element';
+import { Button, Card, CardBody, Notice } from '@wordpress/components';
+import { PLUGINS_STORE_NAME, OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { useDispatch } from '@wordpress/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import Visa from './cards/visa.js';
+import MasterCard from './cards/mastercard.js';
+import Amex from './cards/amex.js';
+import DinersClub from './cards/diners.js';
+import Discover from './cards/discover.js';
+import JCB from './cards/jcb.js';
+import UnionPay from './cards/unionpay.js';
+import './style.scss';
+
+/**
+ * The available experiment treatments.
+ */
+enum Treatment {
+	/**
+	 * Treatment A: Create a subscription product then WCPay onboarding.
+	 */
+	A = 'A',
+
+	/**
+	 * Treatment B: WCPay onboarding then create a subscription product.
+	 */
+	B = 'B',
+}
+
+declare global {
+	interface Window {
+		wcWcpaySubscriptions: {
+			newSubscriptionProductUrl: string;
+			onboardingUrl: string;
+			noThanksUrl: string;
+			dismissOptionKey: string;
+			experimentAssignment: Treatment;
+		};
+	}
+}
+
+const {
+	newSubscriptionProductUrl,
+	onboardingUrl,
+	noThanksUrl,
+	dismissOptionKey,
+	experimentAssignment,
+} = window.wcWcpaySubscriptions;
+
+type setHasErrorFunction = React.Dispatch< React.SetStateAction< boolean > >;
+
+const ErrorNotice = () => {
+	return (
+		<Notice
+			className="wcpay-empty-subscriptions__error"
+			status="error"
+			isDismissible={ false }
+		>
+			{ createInterpolateElement(
+				__(
+					'Installing WooCommerce Payments failed. To continue with WooCommerce Payments built-in subscriptions functionality, please install <a>WooCommerce Payments</a> manually.',
+					'woocommerce'
+				),
+				{
+					a: (
+						// eslint-disable-next-line jsx-a11y/anchor-has-content
+						<a
+							href="https://wordpress.org/plugins/woocommerce-payments/"
+							target="_blank"
+							rel="noreferrer"
+						/>
+					),
+				}
+			) }
+		</Notice>
+	);
+};
+
+const NoThanksButton = () => {
+	const [ isNoThanksClicked, setIsNoThanksClicked ] = useState( false );
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
```

Comment:
Just want to call out that we will hopefully deprecate the options API and datastores in the future.

However, we don't have a replacement API in place yet, so no changes necessary here right now.